index.html

```diff
@@ -251,6 +394,517 @@ <h4>JSON-LD*-specific Terms</h4>
+        <div class="selectors">
+          <button class="selected input" data-selects="compacted">Compacted (Input)</button>
+          <button data-selects="expanded">Expanded</button>
+          <button data-selects="flattened">Flattened</button>
+          <button data-selects="turtle">Turtle*</button>
+        </div>
+        <pre class="compacted input selected"
+             data-transform="updateExample"
+             data-options="rdfstar=true">
+        <!--
+          {
+            "@context": {
+              "@base": "http://example.org/",
+              "@vocab": "http://example.org/",
+              "claims": {"@type": "@id"},
+              "claimedBy": {"@reverse": "http://example.org/claims", "@type": "@id"}
+            },
+            "@id": "bob",
+            "knows": {
+              "@id": "alice",
+              "@annotation": {
+                "claims": "alice",
+                "claimedBy": "bob"
+              }
+            }
+          }
+        -->
+        </pre>
+        <pre class="expanded"
+             data-transform="updateExample"
+             data-options="rdfstar=true"
+             data-result-for="Reversing annotations-compacted">
+        <!--
+          [{
+            "@id": "http://example.org/bob",
+            "http://example.org/knows": [{
+              "@id": "http://example.org/alice",
+              "@annotation": [{
+                "http://example.org/claims": [{"@id": "http://example.org/alice"}],
+                "@reverse": {
+                  "http://example.org/claims": [{"@id": "http://example.org/bob"}]
+                }
+              }]
+            }]
+          }]
+        -->
+        </pre>
+        <pre class="flattened"
+             data-transform="updateExample"
+             data-options="rdfstar=true"
+             data-flatten
+             data-result-for="Reversing annotations-compacted">
+        <!--
+          [{
+            "@id": "http://example.org/bob",
+            "http://example.org/knows": [{"@id": "http://example.org/alice"}]
+          }, {
+            "@id": {
+              "@id": "http://example.org/bob",
+              "http://example.org/knows": [{"@id": "http://example.org/alice"}]
+            },
+            "http://example.org/claims": [{"@id": "http://example.org/alice"}]
+          }, {
+            "@id": "http://example.org/bob",
+            "http://example.org/claims": [{
+              "@id": "http://example.org/bob",
+              "http://example.org/knows": [{"@id": "http://example.org/alice"}]
+            }]
+          }]
+        -->
+        </pre>
+        <pre class="turtle"
+             data-content-type="text/turtle"
+             data-transform="updateExample"
+             data-options="rdfstar=true"
+             data-to-rdf
+             data-result-for="Reversing annotations-compacted">
+        <!--
+        @prefix : <http://example.org/> .
+
+        :alice :claims << :bob :jobTitle "Assistant Designer" >> .
```

Comment:
This is not consistent with the JSON-LD versions of the example. It should be:
```suggestion
        :bob :jobTitle "Assistant Designer".
        :bob :claims << :bob :jobTitle "Assistant Designer" >> .
        << :bob :jobTitle "Assistant Designer" >> :claims :alice.
```
but the use of "claims" on both sides of the embedded triples is strange. Maybe the JSON-LD examples should be fixed instead?

Reply:
I think it is consistent with the JSON-LD, because of the `@reverse`. Consider the expanded version of the JSON-LD:

    [{
      "@id": {
        "@id": "http://example.org/bob",
        "http://example.org/jobTitle": [{"@value": "Assistant Designer"}]
      },
      "@reverse": {
        "http://example.org/claims": [{"@id": "http://example.org/alice"}]
      }
    }]

Because the "claims" property is reversed, it's value becomes the subject and the embedded triple in`@id` becomes the object.

This was drawn from the example in the RDF* doc:

```turtle
:employee22 :claims << :employee38 :jobTitle "Assistant Designer" >> .
```

Reply:
There were some errors, but all the examples now validate using the example validation mechanism based on my implementation. See if this addresses your concerns.

There remains an issue with flattening embedded objects containing blank nodes, after which I'll generate the flattening test suite; toRdf should be fairly straightforward after that.

Reply:
Yes, it addresses my concerns.